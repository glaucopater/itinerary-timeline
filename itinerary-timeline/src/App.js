import React from 'react'; 
import ItineraryContainer from './containers/ItineraryContainer/ItineraryContainer';
import { itinerary } from './data';

const App = () => { 
    return (
      <div className="App"> 
        <ItineraryContainer itinerary={itinerary}/>
      </div>
    );
  
}

export default App;
