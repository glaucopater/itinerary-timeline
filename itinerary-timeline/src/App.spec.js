import React from 'react';
import { shallow } from 'enzyme';
import App from './App';  
import fakeProps from 'react-fake-props';

describe('App', () => {
    const props = fakeProps('./src/containers/ItineraryContainer/ItineraryContainer.jsx'); 
    const app = shallow(<App {...props}/>); 
    it('renders correctly', () => {
        expect(app).toMatchSnapshot();
    })
});
