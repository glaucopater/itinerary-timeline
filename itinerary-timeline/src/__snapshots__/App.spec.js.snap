// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ItineraryContainer
        itinerary={
          Object {
            "destination_iata": "MUC",
            "itinerary_id": "3dd59474-f467-42bc-8098-3c1695ab362d",
            "origin_iata": "TXL",
            "segments": Array [
              Object {
                "departure": 1523596800,
                "destination": Object {
                  "type": "AIRPORT",
                  "value": "TXL",
                },
                "origin": Object {
                  "type": "STREET_ADDRESS",
                  "value": "Alte Schönhauser Straße 3, 10119 Berlin",
                },
                "segment_id": "452dfb51-0ac6-4274-b740-af6f89bc6116",
                "type": "GROUND",
              },
              Object {
                "departure": 1523599200,
                "detail": Object {
                  "busyness_index": 10,
                },
                "origin": Object {
                  "type": "AIRPORT",
                  "value": "TXL",
                },
                "segment_id": "39552e6d-6785-4e83-82dc-157a45773b45",
                "type": "AIRPORT",
              },
              Object {
                "departure": 1523602800,
                "destination": Object {
                  "type": "AIRPORT",
                  "value": "MUC",
                },
                "detail": Object {
                  "boarding": 1523601000,
                  "flight_number": "LH2031",
                  "gate": "C12",
                  "seat": "12C",
                },
                "origin": Object {
                  "type": "AIRPORT",
                  "value": "TXL",
                },
                "segment_id": "07f77c9f-7fdf-446b-91d5-52e900f33723",
                "type": "FLIGHT",
              },
            ],
            "traveler": Object {
              "first_name": "John",
              "last_name": "Doe",
              "middle_name": "C.",
            },
          }
        }
      />,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "itinerary": Object {
          "destination_iata": "MUC",
          "itinerary_id": "3dd59474-f467-42bc-8098-3c1695ab362d",
          "origin_iata": "TXL",
          "segments": Array [
            Object {
              "departure": 1523596800,
              "destination": Object {
                "type": "AIRPORT",
                "value": "TXL",
              },
              "origin": Object {
                "type": "STREET_ADDRESS",
                "value": "Alte Schönhauser Straße 3, 10119 Berlin",
              },
              "segment_id": "452dfb51-0ac6-4274-b740-af6f89bc6116",
              "type": "GROUND",
            },
            Object {
              "departure": 1523599200,
              "detail": Object {
                "busyness_index": 10,
              },
              "origin": Object {
                "type": "AIRPORT",
                "value": "TXL",
              },
              "segment_id": "39552e6d-6785-4e83-82dc-157a45773b45",
              "type": "AIRPORT",
            },
            Object {
              "departure": 1523602800,
              "destination": Object {
                "type": "AIRPORT",
                "value": "MUC",
              },
              "detail": Object {
                "boarding": 1523601000,
                "flight_number": "LH2031",
                "gate": "C12",
                "seat": "12C",
              },
              "origin": Object {
                "type": "AIRPORT",
                "value": "TXL",
              },
              "segment_id": "07f77c9f-7fdf-446b-91d5-52e900f33723",
              "type": "FLIGHT",
            },
          ],
          "traveler": Object {
            "first_name": "John",
            "last_name": "Doe",
            "middle_name": "C.",
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ItineraryContainer
          itinerary={
            Object {
              "destination_iata": "MUC",
              "itinerary_id": "3dd59474-f467-42bc-8098-3c1695ab362d",
              "origin_iata": "TXL",
              "segments": Array [
                Object {
                  "departure": 1523596800,
                  "destination": Object {
                    "type": "AIRPORT",
                    "value": "TXL",
                  },
                  "origin": Object {
                    "type": "STREET_ADDRESS",
                    "value": "Alte Schönhauser Straße 3, 10119 Berlin",
                  },
                  "segment_id": "452dfb51-0ac6-4274-b740-af6f89bc6116",
                  "type": "GROUND",
                },
                Object {
                  "departure": 1523599200,
                  "detail": Object {
                    "busyness_index": 10,
                  },
                  "origin": Object {
                    "type": "AIRPORT",
                    "value": "TXL",
                  },
                  "segment_id": "39552e6d-6785-4e83-82dc-157a45773b45",
                  "type": "AIRPORT",
                },
                Object {
                  "departure": 1523602800,
                  "destination": Object {
                    "type": "AIRPORT",
                    "value": "MUC",
                  },
                  "detail": Object {
                    "boarding": 1523601000,
                    "flight_number": "LH2031",
                    "gate": "C12",
                    "seat": "12C",
                  },
                  "origin": Object {
                    "type": "AIRPORT",
                    "value": "TXL",
                  },
                  "segment_id": "07f77c9f-7fdf-446b-91d5-52e900f33723",
                  "type": "FLIGHT",
                },
              ],
              "traveler": Object {
                "first_name": "John",
                "last_name": "Doe",
                "middle_name": "C.",
              },
            }
          }
        />,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "itinerary": Object {
            "destination_iata": "MUC",
            "itinerary_id": "3dd59474-f467-42bc-8098-3c1695ab362d",
            "origin_iata": "TXL",
            "segments": Array [
              Object {
                "departure": 1523596800,
                "destination": Object {
                  "type": "AIRPORT",
                  "value": "TXL",
                },
                "origin": Object {
                  "type": "STREET_ADDRESS",
                  "value": "Alte Schönhauser Straße 3, 10119 Berlin",
                },
                "segment_id": "452dfb51-0ac6-4274-b740-af6f89bc6116",
                "type": "GROUND",
              },
              Object {
                "departure": 1523599200,
                "detail": Object {
                  "busyness_index": 10,
                },
                "origin": Object {
                  "type": "AIRPORT",
                  "value": "TXL",
                },
                "segment_id": "39552e6d-6785-4e83-82dc-157a45773b45",
                "type": "AIRPORT",
              },
              Object {
                "departure": 1523602800,
                "destination": Object {
                  "type": "AIRPORT",
                  "value": "MUC",
                },
                "detail": Object {
                  "boarding": 1523601000,
                  "flight_number": "LH2031",
                  "gate": "C12",
                  "seat": "12C",
                },
                "origin": Object {
                  "type": "AIRPORT",
                  "value": "TXL",
                },
                "segment_id": "07f77c9f-7fdf-446b-91d5-52e900f33723",
                "type": "FLIGHT",
              },
            ],
            "traveler": Object {
              "first_name": "John",
              "last_name": "Doe",
              "middle_name": "C.",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
